### PR TITLE
[custom channels]: refactor AuxTrafficManager to be used for forwarding as well

### PR DESCRIPTION
**Pull request by guggero**
Renames and moves the `TlvTrafficShaper => AuxTrafficManager` and prepares it to be used for forwarding bandwidth checks as well.

### CODE REVIEW

File: htlcswitch/interfaces.go [hunk 1]
```diff
@@ -461,3 +475,36 @@ type htlcNotifier interface {
...
+	// identified by the provided channel ID may have external mechanisms
+	// that would allow it to carry out the payment.
+	ShouldHandleTraffic(cid lnwire.ShortChannelID,
+		fundingBlob fn.Option[tlv.Blob]) (bool, error)
+
+	// PaymentBandwidth returns the available bandwidth for a custom channel
+	// decided by the given channel aux blob and HTLC blob. A return value
+	// of 0 means there is no bandwidth available. To find out if a channel
+	// is a custom channel that should be handled by the traffic shaper, the
+	// HandleTraffic method should be called first.
```
Comment (GeorgeTsagk) [inline htlcswitch/interfaces.go:507]:
```suggestion
	// ShouldHandleTraffic method should be called first.
```

File: htlcswitch/link.go [hunk 2]
```diff
@@ -3401,8 +3405,38 @@ func (l *channelLink) canSendHtlc(policy models.ForwardingPolicy,
...
+
+			if len(blob) > 0 {
+				htlcBlob = fn.Some(blob)
+			}
+
+			return l.AuxBandwidth(amt, originalScid, htlcBlob, ts)
+		},
+	).Unpack()
+	if err != nil {
+		l.log.Warnf("Unable to determine aux bandwidth: %v", err)
```
Comment (GeorgeTsagk) [inline htlcswitch/link.go:3430]: shouldn't this be an `ERR`? given that it leads to a link error right below

Comment (guggero) [reply htlcswitch/link.go]: Yeah, makes sense. Changed.

File: htlcswitch/link.go [hunk 3]
```diff
@@ -3401,8 +3405,38 @@ func (l *channelLink) canSendHtlc(policy models.ForwardingPolicy,
...
+		return NewLinkError(&lnwire.FailTemporaryNodeFailure{})
+	}
+
+	auxBandwidth.WhenSome(func(bandwidth lnwire.MilliSatoshi) {
+		availableBandwidth = bandwidth
+	})
+
 	// Check to see if there is enough balance in this channel.
-	if amt > l.Bandwidth() {
+	if amt > availableBandwidth {
```
Comment (Roasbeef) [inline htlcswitch/link.go:3439]: Isn't `amt` here in units of BTC so this isn't an accurate comparison? Or will `AuxBandwidth` actually map to BTC?

Comment (guggero) [reply htlcswitch/link.go:3439]: `AuxBandwidth` is always in BTC, it's the asset balance converted to sats, using the quote represented by the RFQ ID in the HTLC blob.

**Review by Roasbeef — APPROVED** [review]
LGTM 🦆

**Review by GeorgeTsagk — APPROVED** [review]
LGTM :ninja:

**Pull request merged**